src/subscriptions/SubscriptionsContainer.js

```diff
@@ -5,12 +5,25 @@ import type { GlobalState } from '../types';
 import { getAuth } from '../selectors';
 import boundActions from '../boundActions';
 import SubscriptionsCard from './SubscriptionsCard';
+import { getUnreadStreams } from '../baseSelectors';
+import { queueMarkAsRead } from '../api';
 
 export default connect(
   (state: GlobalState) => ({
     auth: getAuth(state),
     streams: state.streams,
     subscriptions: state.subscriptions,
+    markStreamMessagesRead: (streamId: number) => {
+      const removeIds = [].concat(
```

Comment:
You are making two unnecessary array copies here.
The first one is the `[].concat()` and the second is the `...` in front of `getUnreadStreams`.
'filter' already creates a new array so `getUnreadStreams(state).filter()` is enough.

Reply:
​state.unread.streams is an array of topic objects, which is why I potentially get a nested array which I am flattening here.